### ISSUE TITLE
add support for reference files

### ISSUE THREAD

**Issue by jantimon** [storybookjs/storybook#14735]
**Is your feature request related to a problem? Please describe**

Storybook 6 deprecated the `configure` and the `storiesOf` API.  
This is fine as most use cases can be solved by the `files` list of `main.js`.

In my mono repository project I am using a special loader which loads all `package.json` files.
e.g.:

main.js
```
module.exports = {
   webpackFinal: () => /* add a loader for package.json */,
   stories: ['my/path/**/*.package.json']
}
```

The loader attaches all stories for the given package and prepends the package name to component name.

For example loading the following `package.json` file would add `Button.storries.tsx` and `GhostButton.storries.tsx` to storybook with the prefix `'@ui/button/'`:
![sharp-pond-5s85s_-_CodeSandbox](https://user-images.githubusercontent.com/4113649/116262824-25ae6680-a779-11eb-85ae-ad6595c0f09d.png)

**Describe the solution you'd like**

It would be cool to allow CSF files returning an array e.g.:

```js
import * as Button from './Button.storries.tsx';
import * as GhostButton from './GhostButton.storries.tsx';
export default [
  Button,
  GhostButton,
]
```

**Are you able to assist to bring the feature to reality?**
yes - I could try to build a POC and add it to the loadCsf function: https://github.com/storybookjs/storybook/blob/32c63568d6f7f6ef90afd206405524543d5a7068/lib/core-client/src/preview/loadCsf.ts#L52-L58

**Comment by shilman** [comment]
can you explain the use case in more detail? i don't yet understand why we need this 🙈

**Comment by jantimon** [comment]
Watching globs like `'../**/*.stories.tsx'` can slow down the develop experience a lot for mono repository projects.

In some projects we used the `main.js` file to reference multiple `package.json` files and add a webpack-loader which adds the stories belonging to the package of that `package.json`.  

A simplified dependency graph might look like this:

![Storybook](https://user-images.githubusercontent.com/4113649/116823076-bc689200-ab82-11eb-85e0-9fabc4901a2d.png)

As CSF and MDX don't support multiple `titles` I am a little bit stuck now that `configure` and `addStoryOf` are deprecated

**Comment by shilman** [comment]
Have you considered the following possible solutions?

## Multiple main.js references

In your `.storybook/main.js`

```js
module.exports = {
  stories: [
    '../a-workspace-module/src/**/*.stories.tsx',
    '../another-workspace-module/src/**/*.stories.tsx',
  ]
...
}
```

## Composition

[Storybook composition](https://storybook.js.org/docs/react/workflows/storybook-composition#gatsby-focus-wrapper) allows you to develop multiple storybooks and then efficiently roll them up into a single project storybook.

It was released in 6.0 and still has some rough edges, but I'd be more inclined to spend the development effort to clean up those rough edges than add a new feature to CSF, if composition satisfies your use case.

**Comment by shilman** [comment]
We’re cleaning house! Storybook has changed a lot since this issue was created and we don’t know if it’s still valid. Please open a new issue referencing this one if this is still relevant in SB 7.x.
